example_python_travis.yml

```diff
@@ -1,46 +1,52 @@
-language: python
-sudo: false
-
-# If you need X11:
-services:
-    - xvfb
-
-# This is for X11 too:
-addons:
-    apt:
-      packages:
-        - herbstluftwm
-        - libxkbcommon-x11-0
+version: ~> 1.0
 
 env:
   global:
-    # Doctr deploy key for <org>/<repo>
-    - secure: "<docs build key>"
-    # You only need to specify these if they are different from these values
-    - REQUIREMENTS: requirements.txt
-    - DEV_REQUIREMENTS: dev-requirements.txt
-    - DOCS_REQUIREMENTS: docs-requirements.txt
-    - DOCS_FOLDER: docs
-    # Assign this variable if you want a version menu in your docs
-    - DOCTR_VERSIONS_MENU: 1
-    # Assign this variable to specify options for the python linter
-    - PYTHON_LINT_OPTIONS="path/to/source --verbose"
+    # doctr generated secure variable for documentation upload
+    - secure: "<your secure here>"
+    # enable the usage of versions menu which allow versioning of the docs
+    # pages and not only the master branch
+    - DOCTR_VERSIONS_MENU="1"
+    # Dependency files used to build the documentation (space separated)
+    - DOCS_REQUIREMENTS="dev-requirements.txt requirements.txt"
+    # Options to be passed to flake8 for package linting. Usually this is just
+    # the package name but you can enable other flake8 options via this config
+    - PYTHON_LINT_OPTIONS="my_package"
+
+    # The name of the conda package
+    - CONDA_PACKAGE="my_package"
+    # The folder containing the conda recipe (meta.yaml)
+    - CONDA_RECIPE_FOLDER="conda-recipe"
+    # Extra dependencies needed to run the tests which are not included
+    # at the recipe and dev-requirements.txt. E.g. PyQt
+    - CONDA_EXTRAS="pip pyqt=5 happi"
+    # Requirements file with contents for tests dependencies
+    - CONDA_REQUIREMENTS="dev-requirements.txt"
 
-# Also for X11:
-before_script:
-  # Run the window manager
-  - "herbstluftwm &"
-  - sleep 1
+    # Extra dependencies needed to run the test with Pip (similar to
+    # CONDA_EXTRAS) but for pip
+    - PIP_EXTRAS="PyQt5 happi"
 
-stages:
-  - test
-  - name: deploy
-    if: (branch = master OR tag IS present) AND type != pull_request
+jobs:
```

Comment:
Maybe start out with this block commented? Would be nice to by default test pypi and uncomment if needed.

Reply:
Addressed via https://github.com/pcdshub/pcds-ci-helpers/pull/34/commits/9c24f3c803c480d26d9eed14f7e753b690bc29c0